Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -3031,6 +3040,39 @@ void webkitWebViewPopulateContextMenu(WebKitWebView* webView, const Vector<WebCo
 #endif
         hitTestResult.get(), &returnValue);
 }
+
+/**
+ * webkit_web_view_select_context_menu_item:
+ * @web_view: a #WebKitWebView
+ * @item: a #WebKitContextMenuItem
+ *
+ * Selects a context menu item.
+ *
+ * This function should be called by WPE applications that render their own
+ * context menu UI. When the user selects a menu item, call this function to
+ * tell WebKit which item was selected. WebKit will then execute the
+ * corresponding action (e.g., copy, paste, open link, etc.).
+ *
+ * This function must be called while the context menu is still active (i.e.,
+ * after the #WebKitWebView::context-menu signal was emitted and before the
+ * #WebKitWebView::context-menu-dismissed signal is emitted).
+ *
+ */
+void webkit_web_view_select_context_menu_item(WebKitWebView* webView, WebKitContextMenuItem* item)
```

Comment:
I think this would be nicer if it was a method of the `WebKitContextMenu` class:

```cpp
void webkit_context_menu_select_item(WebKitContextMenu*, WebKitContextMenuItem*);
```

This may require storing the corresponding `WebKitWebView` instance in the `_WebKitContextMenuPrivate` struct, or maybe a reference to the `WebContextMenuProxyWPE` (and then the page can be picked from that).

@carlosgcampos WDYT?